### DISCUSSION TITLE
Unable to remove elastic-agent from Windows Server

### DISCUSSION THREAD

**Discussion by nightstalker117** [Security-Onion-Solutions/securityonion#13784] [2.4]
### Version

2.4.100

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Meets minimum requirements

### CPU

6

### RAM

24

### Storage for /

293GB

### Storage for /nsm

2TB

### Network Traffic Collection

tap

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

I'm attempting to remove an older version of the so-elastic-agent on a backup DC.  I am unable to stop the service nor uninstall it.  

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by reyesj2** [comment]
From an admin command prompt you should be able to uninstall using
```
C:\"Program Files"\Elastic\Agent\elastic-agent.exe uninstall
```

**Comment by kvkamin** [comment]
I once had to uninstall Elastic Agent from several hundred workstations and realized that since Elastic Defend had been deployed, which also installs the Elastic Endpoint service, I needed to remove Elastic Endpoint first, followed by Elastic Agent. I created a PowerShell script that I deployed through our MSP agent to handle both removals in a single step.

https://www.elastic.co/guide/en/security/current/uninstall-agent.html
